### PR TITLE
#567: Parse data kind and general data format into the CMMStudy model

### PR DESCRIPTION
**Pull request by markusjt**
I started by parsing just Data Kind element text values as free texts but expanded it to also include type attribute. Then I re-read the issue and decided to also add General Data Format parsing for DDI-L 3.3 with CV information included.

Not sure how I should have tested General Data Format parsing since it's not available in DDI-L 3.2 yet but the current DDI-L 3.2 and DDI-L 3.3 xml files were checked against the same json so I created a new xml and a new json (_new_fields).

Going on a vacation now so I'll leave this in draft mode for now since I didn't have time to check everything that I wanted or ask if what I've done is correct for what we want at this stage. Parsing General Data Format with CV information might be a bit of a waste right now but I feel like it's what we should be aiming for eventually so when SPs start using DDI-L 3.3, they should be using values from CV.

### CODE REVIEW

File: src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml
Comment (matthew-morris-cessda) [inline src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml:1]: There appears to be significant changes to this file beyond the additions needed for testing purposes

**Review by matthew-morris-cessda — COMMENTED** [review]
Most of this is fine, see comments on src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml

File: src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml
Comment (markusjt) [reply src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml:1]: Forgot to mention about this. The file didn't include the dataKind element required for this issue even though FSD has provided it for a long time already so I figured it wouldn't hurt to replace the whole file with an up-to-date version.

Comment (matthew-morris-cessda) [reply src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml:1]: Replacing the whole file creates large deltas in the PR change view, which made it much harder to spot that the `dataKind` element had been added. It's fine if the tests pass, but simple modifications to existing test files to add necessary elements is easier given that the purpose of these files is to test parser functionality.

Comment (markusjt) [reply src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml:1]: I get that and I can push a commit so the only change is the addition of the new element being tested. I just thought this specific file was also about testing against real metadata, which loses some value if the file is over five years old.

Comment (matthew-morris-cessda) [reply src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml:1]: Probably not worth making a commit to change the file back, so I'll approve the PR as is

Comment (matthew-morris-cessda) [reply src/test/resources/xml/ddi_2_5/oai-fsd_uta_fi-FSD3187.xml:1]: One the PR is out of draft that is

**Comment by markusjt** [comment]
I noticed that I hadn't actually pushed the commit that I talked about in the first comment that has general data format parsing for DDI-L 3.3 before I left for vacation even though I meant to do so.

I marked this ready for review now but I'll probably look at this a bit more tomorrow so I might still find something by myself then. I left the file oai-fsd_uta_fi-FSD3187.xml with all the changes even though the only relevant change for this was the dataKind element addition like we talked about.

**Review by matthew-morris-cessda — APPROVED** [review]

**Pull request merged**